pyspider/scheduler/task_queue.py

```diff
@@ -38,17 +39,36 @@ def __init__(self, taskid, priority=0, exetime=0):
         self.exetime = exetime
 
     def __cmp__(self, other):
-        if self.exetime == 0 and other.exetime == 0:
-            return -cmp(self.priority, other.priority)
-        else:
+        # compare priority first
+        cmp_priority = -cmp(self.priority, other.priority)
+
+        # when two element have the same priority, then compare exetime.
+        # keep tasks in time order.
+        if cmp_priority == 0:
             return cmp(self.exetime, other.exetime)
+        return cmp_priority
 
     def __lt__(self, other):
         return self.__cmp__(other) < 0
 
 
-class PriorityTaskQueue(Queue.Queue):
+class InTimeQueueTask(InQueueTask):
```

Comment:
It's too complex to add a new class here, you could add a new globally auto increasing value for `InQueueTask` and sort by : exetime, priority, this new property.

Reply:
I will try it